controllers/config.go

```diff
@@ -0,0 +1,235 @@
+/*
+Copyright 2025 Red Hat OpenShift Data Foundation.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ghodss/yaml"
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	extv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	odfOperatorConfigMapName = "odf-operator-manager-config"
+)
+
+var (
+	// configMapResorceVersion will keep the local copy of the odf-operator-manager-config configmap resource version
+	configMapResorceVersion string
+	// OlmPkgRecordList will keep the local copy of parsed data from odf-operator-manager-config configmap
+	OlmPkgRecordList = []*OlmPkgRecord{}
+	// CrdToPackageRecordList will keep the local copy of parsed data from OlmPkgRecordList
+	CrdToPackageRecordList = []*CrdToPackageRecord{}
```

Comment:
after going through the file, you are creating empty slice here and assign a new slice after copying the details to an intermediate slice which leaves old slice for gc everytime.

I propose 2 things, one, the structs doesn't nest with pointers and so you can store the structs directly (only one slice to be pedantic), two, update the older slice with intermediate info and cut down the extra space.